File: utilities.py
import re

import requests
from lxml import html

LI_JOB_VIEW_XPATHS = [# Title
    "/html/body/main/section[1]/div/section[2]/div/div[1]/div/h1//text()", # Company
    "/html/body/main/section[1]/div/section[2]/div/div[1]/div/h4/div[1]/span[1]/a//text()", # Location
    "/html/body/main/section[1]/div/section[2]/div/div[1]/div/h4/div[1]/span[2]//text()", # Description
    '//div[contains(@class, "show-more-less-html__markup show-more-less-html__markup--clamp-after-5")]//text()']


def scrapeJobListing(url, xpaths):
    response = requests.get(url)
    tree = html.fromstring(response.content)

    title = tree.xpath(xpaths[0])[0].strip()

    company = tree.xpath(xpaths[1])[0].strip()

    location = tree.xpath(xpaths[2])[0].strip()

    description = tree.xpath(xpaths[3])
    description = "\n".join(description).encode("latin1").decode("utf-8").strip()

    wage = " - ".join(sorted(re.findall(r"(\$\d+.\d{2}|\$\d+)", description))[:2])

    print(title, company, location, wage, description[:80] + "...", sep="\n")
    return description


def scrapeLinkedInProfile(url):
    pass


def writeToDataset():
    pass


def readFromDataset():
    pass


def generateMessage():
    pass


def recruiterSearch():
    # input LinkedIn tags to url
    #  cURL the content, then scrape for profiles
    pass


def editOnDataset():
    pass

File: utilities.js
chrome.runtime.sendMessage(scrapeJobListing());

function scrapeJobListing() {
    const LI_JOB_VIEW_SELECTORS = {
        title: "/html/body/div[6]/div[3]/div[2]/div/div/main/div[2]/div[1]/div/div[1]/div/div/div/div[2]/div/h1",
        company: "/html/body/div[6]/div[3]/div[2]/div/div/main/div[2]/div[1]/div/div[1]/div/div/div/div[1]/div[1]/div/a",
        location: "/html/body/div[6]/div[3]/div[2]/div/div/main/div[2]/div[1]/div/div[1]/div/div/div/div[3]/div/span[1]",
        description: '/html/body/div[6]/div[3]/div[2]/div/div/main/div[2]/div[1]/div/div[4]/article/div/div[1]/div/p[1]'
    };

    let title = document.evaluate(LI_JOB_VIEW_SELECTORS.title, document, null, XPathResult.FIRST_ORDERED_NODE_TYPE, null).singleNodeValue?.innerText.trim() || "Title not found";
    let company = document.evaluate(LI_JOB_VIEW_SELECTORS.company, document, null, XPathResult.FIRST_ORDERED_NODE_TYPE, null).singleNodeValue?.innerText.trim() || "Company not found";
    let location = document.evaluate(LI_JOB_VIEW_SELECTORS.location, document, null, XPathResult.FIRST_ORDERED_NODE_TYPE, null).singleNodeValue?.innerText.trim() || "Location not found";

    let descriptionElement = document.evaluate(LI_JOB_VIEW_SELECTORS.description, document, null, XPathResult.FIRST_ORDERED_NODE_TYPE, null).singleNodeValue;
    let description = descriptionElement ? descriptionElement.innerText.trim() : "Description not found";

    let wageMatches = description.match(/(\$\d+\.\d{2}|\$\d+)/g);
    let wage = wageMatches ? wageMatches.slice(0, 2).sort().join(" - ") : "Wage not found";

    console.log(title, company, location, wage, description.substring(0, 80) + "...");
    return `Added ${title} @ ${company} to your dashboard!`;
}